### DISCUSSION TITLE
v0.0.7

### DISCUSSION THREAD

**Discussion by carson-katri** [carson-katri/dream-textures#257] [Announcements]
This version contains some awesome new features:
* AI Upscaling with Real-ESRGAN
* A Dream Textures render pass in Cycles
* Negative prompts
* Prompt import/export as JSON
* and many bug fixes and improvements

See the [list of guides](https://github.com/carson-katri/dream-textures#usage) for more information on every feature of the addon.

If you want to discuss the addon, join the [Dream Textures Discord server](https://discord.gg/EmDJ8CaWZ7).

# Windows Installation
If you had a previous version of the addon, click `Remove` in Blender to delete it before installing the update. Then **restart Blender** to ensure the old modules are unloaded.

1. Download the file `dream_textures-win32-cuda.7z` from the Assets section below.
2. Due to file size limits on GitHub, the addon `.zip` file is contained in the 7-Zip archive. Download [7-Zip](https://www.7-zip.org/) if you don't have it already.
3. Right click the file and select `7-Zip` > `Extract Here`
4. Open Blender > Preferences > Add-ons > Install..., then select the file `dream_textures-win32-cuda.zip` extracted from the 7z previously. The addon may take a minute to be copied into Blender.
5. Follow the instructions in the addon preferences window to setup the model weights.

# macOS Installation
1. Download the file `dream_textures-darwin-arm64.zip` from the Assets section below.
2. Open Blender > Preferences > Add-ons > Install..., then select the zip file. It may take a minute to be copied into Blender.
3. Follow the instructions in the addon preferences window to setup the model weights.

# Other Platforms Installation
Other platforms are not officially supported by this addon. However, you can install it manually by following the steps under [Contributing](https://github.com/carson-katri/dream-textures#contributing) in the README.

## What's Changed
* Fixed context bug "AttributeError: 'SpaceDopeSheetEditor' object has … by @blastframe in https://github.com/carson-katri/dream-textures/pull/160
* Fix some install issues by @NullSenseStudio in https://github.com/carson-katri/dream-textures/pull/186
* Subprocess IPC improvements by @NullSenseStudio in https://github.com/carson-katri/dream-textures/pull/204
* Fix sampler argument by @carson-katri in https://github.com/carson-katri/dream-textures/pull/230
* Add Real-ESRGAN integration by @carson-katri in https://github.com/carson-katri/dream-textures/pull/233
* UI Refactor by @carson-katri in https://github.com/carson-katri/dream-textures/pull/240
* Add "Dream Textures" render pass to Cycles by @carson-katri in https://github.com/carson-katri/dream-textures/pull/229

**Full Changelog**: https://github.com/carson-katri/dream-textures/compare/0.0.6...0.0.7

<hr /><em>This discussion was created from the release <a href='https://github.com/carson-katri/dream-textures/releases/tag/0.0.7'>v0.0.7</a>.</em>

**Comment by SolusGod** [comment]
O.M.G

**Reply by SoulBreacher** [reply]
so i instaled the previous version after a lot of hassle into my blender 3.4 but after the update that dropped 9 about 10 hours ago of me writing this its just giving me error after error .....nothing is working for me

and before u give me the obvious answers, yes i restarted my pc, i even completely wiped my pc of blender with the BC uninstaller .....still same problem 

hours down the drain

**Comment by mghoniem** [comment]
Hello Apple Silicon MAcBook Pro 2022 
its very slow.. like really slow.. 
<img width="308" alt="Screen Shot 2022-10-08 at 7 33 55 PM" src="https://user-images.githubusercontent.com/11878255/194720285-edf2e774-bf87-4701-ad69-918bb2c945d0.png">

**Reply by carson-katri** [reply]
Are you running with show steps disabled? That’ll speed it up quite a bit. It takes roughly 20 seconds at default settings on M1 Max.

**Reply by Boboknight11** [reply]
For me, it reloads the model every time I try to render a new image, and obviously that causes a massive slowdown. Would love to see this fixed so I can access the new features of this version.

**Reply by mghoniem** [reply]
to create 512 x 512 wood Texture took 31 MIN.. Minutes ... 
M1 Pro not Max... 
Are you running with show steps disabled?  ( what u mean ? how should i do it ?

**Reply by carson-katri** [reply]
Advanced configuration > Show Steps. Turn that off. On my system it takes 4 minutes with it on and 20 seconds with it off.

**Reply by mghoniem** [reply]
ok i tried steps off... same thing.. as i  write this.. the LAPTOP is very very slow its ben 3 min... and NOTHING on the screen 1%

**Reply by carson-katri** [reply]
That's to be expected, when show steps is off, it doesn't show anything while running.

**Reply by carson-katri** [reply]
@Boboknight11 For your issue, make sure you are running with "Automatic" precision to prevent it from reloading every time.

**Reply by mghoniem** [reply]
i did auto. and no steps.. i got. 2% after 6 min!!

**Reply by mghoniem** [reply]
<img width="401" alt="Screen Shot 2022-10-09 at 8 25 07 PM" src="https://user-images.githubusercontent.com/11878255/194773306-cd346495-dcd5-4d5d-ba1a-19a9ffeca6b6.png">

**Reply by carson-katri** [reply]
M1 Pro is probably just slow then. You could try to restart your computer and close all other programs to free up some memory in case it’s using swap.

**Reply by mghoniem** [reply]
What blender ver ur using ?

**Reply by carson-katri** [reply]
3.3 LTS

**Reply by mghoniem** [reply]
Same here.. damn it.. I'm so bummed...
As if it's not seeing any GPU

**Reply by carson-katri** [reply]
You downloaded the `dream_textures-darwin-arm64.zip` file not the cuda one, correct? You can watch Activity Monitor to see if your GPU is being utilized.

also check the Memory tab to see how much of your VRAM it’s using while running.

**Reply by mghoniem** [reply]
Yes I got that file..  the correct one for OSX ..
Will check activity

**Reply by mghoniem** [reply]
> You downloaded the `dream_textures-darwin-arm64.zip` file not the cuda one, correct? You can watch Activity Monitor to see if your GPU is being utilized.
> 
> also check the Memory tab to see how much of your VRAM it’s using while running.

I cleared VRAM.
Closed all programs
Did everything.. still no hope..

**Comment by Narniarelampiando** [comment]
I can't use the "half precision" option, only the "full precision" option, so the image takes a long time to be generated

**Reply by mghoniem** [reply]
ur running MAcBook M1 ?

**Reply by Narniarelampiando** [reply]
It's not a MAC...
these are the errors that appear when choosing the "half precision" option...


RuntimeError: "LayerNormKernelImpl" not implemented for 'Half'

Internal error, see system console for details

**Reply by carson-katri** [reply]
Don’t use half precision, use automatic. That error means your GPU does not support half precision floats. If you want to speed it up, try disabling “Show Steps”.

**Comment by Josiahgil** [comment]
I'm sorry for having to ask this, but could someone help me understand how which file a newb like me is supposed to install if i want the linux version of the dream textures addon?  I saw a pinned comment on youtub saying to install this addon from the "releases tab", which i see has a tag icon. when i arrive, i see it tells me which file to download for windows or macos, but what should i download for linux?  last time, i downloaded the zip called dream_textures.zip, which is why it didn't work, i just want to make sure this time i know which file to download for linux.  I know you guys are doing awesome work, way over my head stuff, and i want to say thanks alot.

**Reply by carson-katri** [reply]
Follow the instructions for “Other Platforms” on the release notes for Linux. It’s a more involved process because every Linux install seemed to work differently, and I do not have a Linux machine to create and test builds on.

**Reply by Josiahgil** [reply]
Thanks i'll do that. I was able to get it to work on windows (i have a dual boot pc, you could try dual boot, or vm if you wanted a linux machine) I only use linux because blender runs at like twice the speed it does in windows. Thanks again.

**Reply by Josiahgil** [reply]
I'm sorry for the newb questions, the instruction says "After cloning the repository, there a few more steps you need to complete to setup your development environment:

    Install submodules:

git submodule update --init --recursive"

Does cloning repository  refer to installing dependencies?

**Reply by carson-katri** [reply]
No it’s referring to downloading a git repository with `git clone`

**Reply by Josiahgil** [reply]
I just learned something new, Thanks alot for you  help.

**Reply by Josiahgil** [reply]
I tried putting "git submodule update --init --recursive" into the terminal, here is the error i get in terminal "fatal: not a git repository (or any of the parent directories): .git"  One results o ngoogle says i need to navigate to the right folder in order for it to work, but i don't know which folder.

**Reply by carson-katri** [reply]
Make sure you are running from inside the dream textures folder

**Reply by Josiahgil** [reply]
okay, i''m sorry, i just realize what you meant by the dream textures folder. Thanks alot

**Reply by Josiahgil** [reply]
I coudn't figure out where to download the addon for blender.  I know where the windows and macos ones are, but which one is for linux?  I mean, where do i get the addon.zip download? thanks

**Reply by carson-katri** [reply]
As mentioned, you need to download the source code and setup manually with the Contributing instructions.

**Reply by Josiahgil** [reply]
Thanks.  Is the zip file supposed to be small like 1.6 mb?

**Reply by carson-katri** [reply]
Yes it’s just the code. The instructions will download all the other stuff that is included in packaged releases for other platforms.

**Reply by Josiahgil** [reply]
okay, thanks alot lol

**Reply by Josiahgil** [reply]
i got an error in the installer saying "Stable diffusion is missing.  You like downloaded the source instead of the release by accident."  i must have made a mistake, but i installed the blender dev in vs code, and so on...

**Reply by carson-katri** [reply]
You need to do the submodule step. You can also join the discord to get help from other users who have gotten it running on windows.

**Reply by Josiahgil** [reply]
okay, i do need that discord, but i can't find a link to it anywhere.  I installed the "git submodule update --init --recursive" into the dream textures folder, and it says it installed, so idk.   You were talking about the dream textures folder created by git clone right? not by the blender addon?

**Reply by carson-katri** [reply]
Git clone downloads the blender addon… you need to place it in blenders folder manually because it’s not a zip.

here’s the discord: https://discord.gg/REQGx73z

**Reply by Josiahgil** [reply]
Thanks alot!

**Reply by Anyeos** [reply]
I did it as (I don't want to duplicate info that is why I linked it):
https://github.com/carson-katri/dream-textures/discussions/231#discussioncomment-3900798

**Comment by mghoniem** [comment]
im trying to install the Addon on Intel BAsed MAC... what file should  i download ?  when i downloaded Darwin-Arm64 i get error worng Archt. 

OSError('dlopen(/Users/Mghoniem/Library/Application Support/Blender/3.0/scripts/addons/dream_textures/.python_dependencies/torch/lib/libtorch_global_deps.dylib, 10): no suitable image found.  Did find:\n\t/Users/Mghoniem/Library/Application Support/Blender/3.0/scripts/addons/dream_textures/.python_dependencies/torch/lib/libtorch_global_deps.dylib: mach-o, but wrong architecture\n\t/Users/Mghoniem/Library/Application Support/Blender/3.0/scripts/addons/dream_textures/.python_dependencies/torch/lib/libtorch_global_deps.dylib: mach-o, but wrong architecture')


Should u download WIN file ?

**Comment by mghoniem** [comment]
OK i installed 
Blender 3.0
OSX Catalina
MAcBook PRo INTEL

when i come to use the plugin i get an error  ( wrong architecture')
i tried installing both version 
WIN and Zip DarwinArm64..
i dont know what to do,,

BElow Error 
OSError('dlopen(/Users/Mghoniem/Library/Application Support/Blender/3.0/scripts/addons/dream_textures/.python_dependencies/torch/lib/libtorch_global_deps.dylib, 10): no suitable image found.  Did find:\n\t/Users/Mghoniem/Library/Application Support/Blender/3.0/scripts/addons/dream_textures/.python_dependencies/torch/lib/libtorch_global_deps.dylib: mach-o, but wrong architecture\n\t/Users/Mghoniem/Library/Application Support/Blender/3.0/scripts/addons/dream_textures/.python_dependencies/torch/lib/libtorch_global_deps.dylib: mach-o, but wrong architecture')

**Reply by carson-katri** [reply]
Only Apple Silicon macs are supported.

**Reply by psyfire25** [reply]
thats a shame

**Comment by psyfire25** [comment]
Every time i try and download I get a network error. need authorisation?

**Reply by carson-katri** [reply]
You do not need to download dependencies in 0.0.7. Unless there is some other error you are referring to?

**Comment by ghost** [comment]
Hello! I cannot get rid of this error, I have tried with different versions of Blender from 3.0 to 3.3.
Numpy is installed but it seemed to be a different version ?
I am using WIndows11.
Numpy version on my system is : 1.23.4
Numpy version on my blender is : 1.22.0
Is where the problem comes from ?
Maybe you'd have some advices please ?

C:\Users\xxx\AppData\Roaming\Python\Python310\site-packages\torchvision\models\detection\anchor_utils.py:63: UserWarning: Failed to initialize NumPy: module compiled against API version 0x10 but this version of numpy is 0xf (Triggered internally at  ..\torch\csrc\utils\tensor_numpy.cpp:68.)
  device: torch.device = torch.device("cpu"),
Traceback (most recent call last):
  File "C:\Users\xxx\AppData\Roaming\Blender Foundation\Blender\3.1\scripts\addons\dream_textures\generator_process.py", line 373, in main
    generator.load_model()
  File "C:\Users\xxx\AppData\Roaming\Blender Foundation\Blender\3.1\scripts\addons\dream_textures\stable_diffusion\ldm\generate.py", line 697, in load_model
    model = self._load_model_from_config(self.config, self.weights)
  File "C:\Users\xxx\AppData\Roaming\Blender Foundation\Blender\3.1\scripts\addons\dream_textures\stable_diffusion\ldm\generate.py", line 839, in _load_model_from_config
    model = instantiate_from_config(c.model)
  File "C:\Users\xxx\AppData\Roaming\Blender Foundation\Blender\3.1\scripts\addons\dream_textures\stable_diffusion\ldm\util.py", line 90, in instantiate_from_config
    return get_obj_from_str(config['target'])(
  File "C:\Users\xxx\AppData\Roaming\Blender Foundation\Blender\3.1\scripts\addons\dream_textures\stable_diffusion\ldm\models\diffusion\ddpm.py", line 641, in __init__
    super().__init__(conditioning_key=conditioning_key, *args, **kwargs)
  File "C:\Users\xxx\AppData\Roaming\Blender Foundation\Blender\3.1\scripts\addons\dream_textures\stable_diffusion\ldm\models\diffusion\ddpm.py", line 141, in __init__
    self.register_schedule(
  File "C:\Users\xxx\AppData\Roaming\Blender Foundation\Blender\3.1\scripts\addons\dream_textures\stable_diffusion\ldm\models\diffusion\ddpm.py", line 736, in register_schedule
    super().register_schedule(
  File "C:\Users\xxx\AppData\Roaming\Blender Foundation\Blender\3.1\scripts\addons\dream_textures\stable_diffusion\ldm\models\diffusion\ddpm.py", line 171, in register_schedule
    betas = make_beta_schedule(
  File "C:\Users\xxx\AppData\Roaming\Blender Foundation\Blender\3.1\scripts\addons\dream_textures\stable_diffusion\ldm\modules\diffusionmodules\util.py", line 59, in make_beta_schedule
    return betas.numpy()
RuntimeError: Numpy is not available

**Comment by Blendeon** [comment]
Doesn´t work for me. :( When I try to enable the Add-on i get this message: 

Traceback (most recent call last):
  File "C:\Program Files (x86)\Steam\steamapps\common\Blender\3.3\scripts\modules\addon_utils.py", line 333, in enable
    mod = __import__(module_name)
  File "C:\Users\leonb\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\dream_textures\__init__.py", line 30, in <module>
    from .render_pass import register_render_pass, unregister_render_pass
  File "C:\Users\leonb\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\dream_textures\render_pass.py", line 8, in <module>
    from .operators.dream_texture import dream_texture
ImportError: cannot import name 'dream_texture' from 'dream_textures.operators.dream_texture' (C:\Users\leonb\AppData\Roaming\Blender Foundation\Blender\3.3\scripts\addons\dream_textures\operators\dream_texture.py)

followed the installation guide for 0.07 exactly and tried several times. Anyone has a clue what went wrong?

**Reply by NullSenseStudio** [reply]
You likely just need to restart Blender.

**Reply by Anyeos** [reply]
Same happened to me when I updated with a pull on an older version. Try to clone a new repository, not to pull over an existing one.

Delete the old one under Blender addons and copy a new fresh one.

Ah, being on Windows you can simply download the release binary and install that. But don't forget to delete the old folder of the addon on Blender (or simply uninstall/remove it from Blender).

**Comment by slowday999** [comment]
![image](https://user-images.githubusercontent.com/116126912/196564275-58d36a2f-e484-4653-a19e-05cca0d844c6.png)
help

**Reply by slowday999** [reply]
Blender 3.3.1 LTS

**Reply by carson-katri** [reply]
Your GPU did not have enough VRAM. Please try reducing the image size

**Reply by slowday999** [reply]
RuntimeError('CUDA out of memory. Tried to allocate 2.00 MiB (GPU 0; 4.00 GiB total capacity; 3.19 GiB already allocated; 1.70 MiB free; 3.24 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_CUDA_ALLOC_CONF')

I set 128x128 but still nothing happens =(
The error says that for some reason only 2 MiB is allocated, or I'm wrong.
![image](https://user-images.githubusercontent.com/116126912/196820629-351121c6-cd46-4291-a789-3e65d2cd3b5f.png)

**Reply by slowday999** [reply]
please help

**Comment by rionovriandra** [comment]
Hi i've installed the add on on Blender 3.3.1 LTS but got this error when I'm trying to generate the prompt :

ImportError('Failed to load PyTorch C extensions:\n    It appears that PyTorch has loaded the `torch/_C` folder\n    of the PyTorch repository rather than the C extensions which\n    are expected in the `torch._C` namespace. This can occur when\n    using the `install` workflow. e.g.\n        $ python setup.py install && python -c "import torch"\n\n    This error can generally be solved using the `develop` workflow\n        $ python setup.py develop && python -c "import torch"  # This should succeed\n    or by running Python from a different directory.')

Please help ;(

**Reply by carson-katri** [reply]
Make sure you installed the correct file, for Windows it is `dream_textures-win32-cuda.7z`.

**Comment by rionovriandra** [comment]
Yes, sorry, i didn't pay attention for the OS.

**Comment by rionovriandra** [comment]
Hi carson, sorry for being a trouble...I've install the right version but when I run it, it comes to "process closed unexpectedly", with this on the system console :

![sgbfghf](https://user-images.githubusercontent.com/116143984/196868498-74144ee5-cd38-429e-86b7-13540e11b9c4.jpg)

**Comment by GUEVARDO** [comment]
Hi friends, I already installed the addon, but once in blender I press ´´ok´´ to generate the image, and nothing happens, I mean the laptop gets a little bit laggy for about two minutes and then it goes back to normal, but no image is generated. My laptop has nvidia gtx1650, 8gb ram and core i5 10300h. I already changed some parameters like resolution, seed and CFG scale, but nothing happens. any idea what´s wrong?... sorry for my bad english, thanks

**Reply by slowday999** [reply]
maybe you have the same problem as me.
I'm still waiting for a response from the author of the addon on the post just above
